app/models/miq_worker.rb

```diff
@@ -209,7 +209,13 @@ def self.fetch_worker_settings_from_server(miq_server, options = {})
         # Clean up the configuration values in a format like "30.seconds"
         unless raw
           settings.keys.each do |k|
-            settings[k] = settings[k].to_i_with_method if settings[k].respond_to?(:to_i_with_method) && settings[k].number_with_method?
+            if settings[k].kind_of?(String)
+              if settings[k].number_with_method?
+                settings[k] = settings[k].to_i_with_method
+              elsif settings[k] =~ /\A\d*(.\d*)?\z/ # case where int/float saved as string
```

Comment:
Should the first one be `\d+` instead of `\d*`, otherwise I think `.123` will pass (though technically it will work as `".123".to_i # => 0`

Reply:
Actually, thinking about it more, maybe `\d+` for both because `123.` will also pass the regex.

Reply:
Thanks @Fryguy , I agree.  Updated.  👍